Userland/Libraries/LibWeb/HTML/FormControlInfrastructure.cpp

```diff
@@ -40,4 +46,140 @@ WebIDL::ExceptionOr<Entry> create_entry(JS::Realm& realm, DeprecatedString const
     };
 }
 
+// https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#constructing-the-form-data-set
+WebIDL::ExceptionOr<Optional<HashMap<DeprecatedString, Vector<Variant<JS::Handle<FileAPI::File>, DeprecatedString>>>>> construct_entry_list(JS::Realm& realm, HTMLFormElement& form)
+{
+    // 1. If form's constructing entry list is true, then return null.
+    if (form.constructing_entry_list())
+        return Optional<HashMap<DeprecatedString, Vector<Variant<JS::Handle<FileAPI::File>, DeprecatedString>>>> {};
+
+    // 2. Set form's constructing entry list to true.
+    form.set_constructing_entry_list(true);
+    ScopeGuard scope_guard([&] {
+        form.set_constructing_entry_list(false);
+    });
```

Comment:
The only spot the form's constructing entry list is set the false is at the end.
Do we actually need to reset this in case of internal errors (perhaps ignoring the OOM ones)